homeassistant/components/zwave_js/climate.py

```diff
@@ -526,10 +506,19 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
         # Dry and Fan preset modes are deprecated as of 2023.8
         # Use Dry and Fan HVAC modes instead
         if preset_mode_value in (ThermostatMode.DRY, ThermostatMode.FAN):
-            LOGGER.warning(
-                "Dry and Fan preset modes are deprecated and will be removed in a future release. "
-                "Use the corresponding Dry and Fan HVAC modes instead"
```

Comment:
I would leave the warning, also add information about the HA release when this support is to be removed.

Reply:
I'm ok with keeping the warning or removing it.

Reply:
I've added back the warning.

Re information about the HA release when this support is to be removed, I'm not sure when it will happen, that's why I haven't included any HA release info. @raman325 do we have a target release to apply the breaking change?

Reply:
I forgot about this comment.

We should keep it for at least half a year. I think we should comment both when we introduced the deprecation and when we plan to remove it.

Current release cycle is 2023.8.0.
Remove earliest in 2024.2.0.

Reply:
@MartinHjelmare - Is it mandatory that we add a breaking change release in this PR or we can leave it open until we agree on a date?

Reply:
I think we can leave it open for now. We have commented when it was added. That's good.